src/main/scala/org/allenai/pipeline/Pipeline.scala

```diff
@@ -35,15 +33,35 @@ trait Pipeline extends Logging {
     runPipelineReturnResults(title, persistedSteps.toSeq)
   }
 
-  def runOne[T](target: Producer[T]) =
-    runOnly(target.stepInfo.className, List(target): _*).toList(0).asInstanceOf[T]
+  def runOne[T](target: Producer[T], tempOutput: Option[String] = None) = {
+    val targetWithOverriddenLocation: Producer[T] =
+      tempOutput match {
+        case Some(tmp) =>
+          if (target.isInstanceOf[PersistedProducer[T, _ <: FlatArtifact]]) {
```

Comment:
Are you sure instanceOf can handle all these cases at runtime? 

I'm thinking maybe it is possible to have PersistedProducer.redirect() { new PP(step) } which just works regardless of the Artifact style?